src/sections/layer_and_mask_information_section/mod.rs

```diff
@@ -263,11 +271,22 @@ fn read_layer_record(cursor: &mut PsdCursor) -> Result<LayerRecord, Error> {
     // until we stop seeing them.
     while cursor.peek_4()? == SIGNATURE_EIGHT_BIM || cursor.peek_4()? == SIGNATURE_EIGHT_B64 {
         let _signature = cursor.read_4()?;
-        let _key = cursor.read_4()?;
+        let _key = Vec::from(cursor.read_4()?);
         let additional_layer_info_len = cursor.read_u32()?;
-        cursor.read(additional_layer_info_len)?;
+
+        if _key.as_slice() == KEY_UNICODE_LAYER_NAME {
+            let length = cursor.read_u32()?;
+            let label_name = cursor.read(additional_layer_info_len - 4)?;
+
+            name = String::from_utf16(
+                &u8_slice_to_u16(label_name).as_slice()[..length as usize]
+            )?;
+        } else {
+            cursor.read(additional_layer_info_len)?; // just skip all data
```

Comment:
Mind making this more clear that we're only skipping because these aren't yet implemented?

i.e. `TODO: Skipping other keys until we implement parsing for them`